src/maxminddb.c

```diff
@@ -1384,7 +1384,10 @@ LOCAL int decode_one(MMDB_s *mmdb, uint32_t offset,
 {
     const uint8_t *mem = mmdb->data_section;
 
-    if (offset + 1 > mmdb->data_section_size) {
+    // We subtract rather than add as it possible that offset + 1
+    // could overflow for a corrupt database while an underflow
+    // from data_section_size - 1 should not be possible.
+    if (offset > mmdb->data_section_size - 1) {
```

Comment:
I wonder if we should add a check that `data_section_size` is not zero (possibly where we initialize it) causing the `- 1` to wrap around? It sounds like it should never be the case, but a check could give more confidence.

I see you do checks that the `data_section_size` is not too small in a few places in this changeset. Maybe we should do that in all of them? Possibly there is a minimum already enforced somewhere that I missed.